cloudbuild/python-scripts/utils/utils.py

```diff
@@ -0,0 +1,249 @@
+          current_timestamp: The current timestamp,
+          the base for calculating the offset.
+        """
+        offset_timestamp = current_timestamp + datetime.timedelta(
+            hours=partition_number
+        )
+
+        # Write the specified number of rows.
+        for _ in range(number_of_rows_per_batch):
+            writer.append({
+                'name': generate_string(),
+                'number': generate_long(),
+                'ts': generate_timestamp(offset_timestamp),
+            })
+
+    def write_avros(
+        self, avro_file_local_identifier, partition_number, current_timestamp
+    ):
+        """Method to generate fake records for BQ table.
+
+        Args:
+          avro_file_local_identifier: The name of the avro file to be used by the
+            current thread.
+          partition_number: The partition number being created - only relevant in
+            partitioned table creation.
+          current_timestamp: Timestamp, one hour within which timestamp entries need
+            to be generated.
+
+        Raises:
+          RuntimeError: When invalid table_type is provided.
+        """
+
+        writer = avro.datafile.DataFileWriter(
+            open(avro_file_local_identifier, 'wb'),
+            avro.io.DatumWriter(),
+            self.schema,
+        )
+        self.write_rows(
+            self.number_of_rows_per_batch,
+            writer,
+            partition_number,
+            current_timestamp,
+        )
+        writer.close()
+
+    def transfer_avro_rows_to_bq_table(self, avro_file_local_identifier):
+        """Method to load the created rows to BQ.
+
+        Args:
+          avro_file_local_identifier: The name of the avro file to
+            be used by the current thread.
+        """
+        client = bigquery.Client()
+
+        job_config = bigquery.LoadJobConfig(
+            create_disposition=bigquery.CreateDisposition.CREATE_IF_NEEDED,
+            write_disposition=bigquery.WriteDisposition.WRITE_APPEND,
+            source_format=bigquery.SourceFormat.AVRO,
+            use_avro_logical_types=True,
+        )
+
+        with open(avro_file_local_identifier, 'rb') as source_file:
+            job = client.load_table_from_file(
+                source_file, self.table_id, job_config=job_config
+            )
+        job.result()
+
+    def delete_local_file(self, avro_file_local_identifier):
+        os.remove(avro_file_local_identifier)
+
+    def avro_to_bq_with_cleanup(
+        self,
+        avro_file_local_identifier,
+        partition_number=0,
+        current_timestamp=datetime.datetime.now(datetime.timezone.utc),
+    ):
+        self.write_avros(
+            avro_file_local_identifier, partition_number, current_timestamp
+        )
+        self.transfer_avro_rows_to_bq_table(avro_file_local_identifier)
+        self.delete_local_file(avro_file_local_identifier)
+
+
+class ArgumentInputUtils:
```

Comment:
This is ok, but we can also use argparse: https://docs.python.org/3/library/argparse.html

Reply:
+1. Seems more pythonic